### DISCUSSION TITLE
Strange issue with runner & openbox

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2134] [General]
On both machines (arch/debian) runner _sometimes_ opens results or menu in the top left corner for some days now. It happens often but can not be triggered.
Running Picom and openbox-patched. 

Already tried with no result:

* Default rc.xml
* changed settings for window placement
* openbox from repository
* disabled picom
* different styles
* reboot ;)

It's quite annoying because search results aren't there where I look...

![schermata-11-22-12-07-01](https://user-images.githubusercontent.com/10681413/142851091-222a55ac-480e-4a16-81ca-823efb09b63b.png)

**Comment by tsujan** [comment]
Perhaps related to https://github.com/lxqt/lxqt/discussions/2105.

If disabling picom makes no difference, the bug should be in Openbox.

A program whose development is stopped will show problems sooner or later, unless someone does something about it. It reminds me of Compiz, which was continued as "compiz-reloaded".

**Comment by tsujan** [comment]
I also remembered https://github.com/lxqt/pcmanfm-qt/issues/1033.

What happens if you add the following line immediately *before* https://github.com/lxqt/lxqt-runner/blob/b7814dddd3b5d6a8fe7acba1f6740b2e82a19eda/dialog.cpp#L356:

```c++
move(rect.topLeft());
```

**Reply by stefonarch** [reply]
> What happens if you add the following line immediately _before_ https://github.com/lxqt/lxqt-runner/blob/b7814dddd3b5d6a8fe7acba1f6740b2e82a19eda/dialog.cpp#L356:

It happens too. One way to trigger it almost always is opening an lxqt-config window, like appearance, and then use the runner.

I still suspect picom in some way, because openbox is the same for ages, no updates. But just now I saw it happen with compton too.
A short test with picom+xfwm4 showed no issue.

**Reply by tsujan** [reply]
> because openbox is the same for ages, no updates

Openbox may be but the rest of your system — X11 included — isn't.

**Reply by stefonarch** [reply]
On debian there were zero updates for X11  last weeks, only one browser and element (matrix). Something has triggered it but what?

**Reply by tsujan** [reply]
`cairo`, `mesa`?

**Reply by stefonarch** [reply]
```
2021-11-03 13:50:35 status installed libegl1-mesa:amd64 20.3.5-1
2021-11-03 13:50:35 status installed libgl1-mesa-glx:amd64 20.3.5-1

2021-11-07 09:41:34 status installed libcairo-script-interpreter2:amd64 1.16.0-5
2021-11-07 09:41:34 status installed libcairo2-dev:amd64 1.16.0-5

```

I noticed it only 2 or 3 days ago.
I can trigger it too (always opening an lxqt-config-* window first) on the VM with ubuntu+git I used for the screenshots, but it appears only for a fraction of seconds in the corner.

**Reply by tsujan** [reply]
> I noticed it only 2 or 3 days ago.

Then you may be able to examine the list of updates in the recent week and find the packages that may affect graphics in any way.

I tested with my other computer, whose last update was in ~2021-11-22~ 2021-10-20, and I couldn't reproduce the issue with Openbox+picom. If I upgrade it and see the issue, the number of packages will be so great that I won't be able to find anything.

**Reply by stefonarch** [reply]
Just looked on my PC running arch, last update was 10 days ago,  xorg-server (1.20.13-3 -> 21.1.1-3 is among it. But sure debian has an older version.

**Reply by tsujan** [reply]
After https://github.com/lxqt/lxqt/discussions/2134#discussioncomment-1682592, I don't think it's about updateds anymore.

**Comment by tsujan** [comment]
Good news and bad news:

On my other computer, with packages from the last month, I tested with Openbox + picom again but, this time, I changed the position of Runner from top to center. Then, the config menu didn't show up at first. Only after I pressed the button several times, it appeared at a wrong position. But, after I closed Runner and reopened it, it always appeared at the correct position. I tested more than 20 times and even rebooted; the position of the menu was correct.

Then, I changed the position of Runner from center to top and the same thing happened: *Only the first time the dialog was shown*, the config menu appeared at the wrong position.

Opening of another window had no role in what I saw.

I repeated the same experiment with KWin; everything was fine.

So, there should be a bug in Openbox. The bad news is that it'll never be fixed.

**Reply by stefonarch** [reply]
Can repeat this in the VM with ubuntu and LXQt git. Same in another VM with debian testing and LXQt 0.16
Search results will show up shortly only in wrong position and than jump in the right place, but only once.

But here it's worse as it affects search results display many times, and I'm quite sure that it starts only after opening some lxqt-config windows.

**Reply by stefonarch** [reply]
Gave up finding the reason, switched to xfwm4, picom and bigsur theme on both machines.

**Reply by tsujan** [reply]
> Gave up finding the reason

The reason is Openbox itself.

**Reply by stefonarch** [reply]
yeah, but why is it on my machines persistent and doesn't jump in the right position, as tested by you and in the VM I've checked

**Reply by tsujan** [reply]
Since I've never studied the code of Openbox, I can't answer that.

But I saw weird things about Openbox years ago, when it was developed. Changing of the positions, as you described here, was among them: When LXQt started, panels appeared at the center for a moment and then, they went to their appropriate positions. That didn't happen with compiz-reloaded, whose original code was very old too.

**Comment by tsujan** [comment]
To make sure that it isn't about old X11 codes, I compiled and installed compiz-reloaded from AUR. No problem with Runner or menus.

BTW, compiz-reloaded still works like a charm with Intel + modesetting + DRI2 (`export LIBGL_DRI3_DISABLE=1`).

**Reply by stefonarch** [reply]
Good. I tiried some time again compiz, but probably not "reloaded", it was like going back to the time of the first landing on the moon, whit the famous cube everyone was so excited about.

The bug as   reproduced  by you (and me in VM) was always around but doesnt affect usability, but what triggered that unusable behavior in both my setups is not clear. Anyway, xfwm4 does it's job and picom too, the internal compositor lacks my favourite, "darken inactive".

**Reply by tsujan** [reply]
> it was like going back to the time of the first landing on the moon

Yes, it's a nostalgic experience. I'll continue using compiz-reloaded (0.8.18) instead of KWin for a while, especially because they removed the desktop cube from KWin.

Wayfire is its reincarnation for Wayland.
